src/chrome/handleStateSaving.js

```diff
@@ -34,10 +34,11 @@ const handleStateSaving = (store, chrome) => {
   listenStoreChanges(store, handleUpdate);
 
   return new Promise(resolve => {
-    chrome.storage.local.get(['workerState'], data => {
-      if (data.workerState === 'ready') {
+    chrome.storage.local.get(['workerState', 'reload'], data => {
+      if (data.reload && data.workerState === 'ready') {
         store.dispatch(updateWorkerState(data.workerState));
       }
+      chrome.storage.local.set({ reload: false });
```

Comment:
Also here, the second argument should just be `resolve` I think (instead of evoking it later).